internal/dot/graph.go

```diff
@@ -42,3 +44,29 @@ type Node struct {
 	Optional bool
 	Group    string
 }
+
+// String returns the string representation of a node so the different
+// constructors can refer to the same node. We omit information on the optional
+// field since the same type can be optional to one constructor and required
+// for another.
+func (n *Node) String() string {
+	if n.Name != "" {
+		return fmt.Sprintf("%v[name=%v]", n.Type, n.Name)
+	} else if n.Group != "" {
+		return fmt.Sprintf("%v[group=%v]", n.Type, n.Group)
+	}
+
+	return n.Type
+}
+
+// Attributes composes and returns a string to style the sublabels when
+// visualizing graph.
+func (n *Node) Attributes() string {
+	if n.Name != "" {
+		return fmt.Sprintf("<BR /><FONT POINT-SIZE=\"10\">Name: %v</FONT>", n.Name)
+	} else if n.Group != "" {
+		return fmt.Sprintf("<BR /><FONT POINT-SIZE=\"10\">Group: %v</FONT>", n.Group)
```

Comment:
Is the `<BR/>` part of the "attributes" of this node?

Reply:
The string produced is to be appended to the labels in the graph. I'm not sure if `<BR />` should be included here, but we only want a newline if the label has a name or group sub-label. Maybe I can rename the method to something like `Sublabel`?